aws/data_source_aws_ec2_instance_type_test.go

```diff
@@ -0,0 +1,101 @@
+					resource.TestCheckResourceAttr(resourceMetal, "supported_usages_classes.#", "2"),
+					resource.TestCheckResourceAttr(resourceMetal, "supported_usages_classes.0", "on-demand"),
+					resource.TestCheckResourceAttr(resourceMetal, "supported_usages_classes.1", "spot"),
+					resource.TestCheckResourceAttr(resourceMetal, "sustained_clock_speed", "3.1"),
+					resource.TestCheckResourceAttr(resourceMetal, "total_instance_storage", "60000"),
+					resource.TestCheckResourceAttr(resourceMetal, "instance_disks.#", "1"),
+					resource.TestCheckResourceAttr(resourceMetal, "instance_disks.0.count", "8"),
+					resource.TestCheckResourceAttr(resourceMetal, "instance_disks.0.size", "7500"),
+					resource.TestCheckResourceAttr(resourceMetal, "instance_disks.0.type", "ssd"),
+					resource.TestCheckResourceAttr(resourceGpu, "total_gpu_memory", "4096"),
+					resource.TestCheckResourceAttr(resourceGpu, "hypervisor", "xen"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.#", "1"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.count", "1"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.memory_size", "4096"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.manufacturer", "NVIDIA"),
+					resource.TestCheckResourceAttr(resourceGpu, "gpus.0.name", "K520"),
+					resource.TestCheckResourceAttr(resourceGpu, "valid_threads_per_core.#", "2"),
+					resource.TestCheckResourceAttr(resourceGpu, "valid_threads_per_core.0", "1"),
+					resource.TestCheckResourceAttr(resourceGpu, "valid_threads_per_core.1", "2"),
+					resource.TestCheckResourceAttr(resourceGpu, "default_threads_per_core", "2"),
+					resource.TestCheckResourceAttr(resourceGpu, "default_cores", "4"),
+					resource.TestCheckResourceAttr(resourceGpu, "default_vcpus", "8"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.#", "1"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.name", "Virtex UltraScale (VU9P)"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.manufacturer", "Xilinx"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.count", "1"),
+					resource.TestCheckResourceAttr(resourceFpga, "fpgas.0.memory_size", "65536"),
+					resource.TestCheckResourceAttr(resourceFpga, "total_fpga_memory", "65536"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "accelerators.#", "1"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "accelerators.0.count", "1"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "accelerators.0.name", "Inferentia"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "accelerators.0.manufacturer", "AWS"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "valid_cores.#", "1"),
+					resource.TestCheckResourceAttr(resourceAccelerator, "valid_cores.0", "2"),
+				),
+			},
+		},
+	})
+}
+
+const testAccDataSourceEc2InstanceType = `
+data "aws_ec2_instance_type" "metal" {
+	instance_type="i3en.metal"
```

Comment:
```suggestion
  instance_type="i3en.metal"
```

Formatting for the HCL must conform to `terraform fmt` which uses 2 spaces instead of a tab for indent here.